src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.cs

```diff
@@ -226,22 +214,7 @@ public override bool IsFinalSpanSemanticallyValidSpan(
             if (body.CloseBraceToken != GetLastTokenInSelection().GetNextToken(includeZeroWidth: true))
                 return false;
 
-            // alright, for these constructs, it must be okay to be extracted
-            switch (container.Kind())
-            {
-                case SyntaxKind.AnonymousMethodExpression:
-                case SyntaxKind.SimpleLambdaExpression:
-                case SyntaxKind.ParenthesizedLambdaExpression:
-                    return true;
-            }
-
-            // now, only method is okay to be extracted out
-            if (body.Parent is not MethodDeclarationSyntax method)
-                return false;
-
-            // make sure this method doesn't have return type.
-            return method.ReturnType is PredefinedTypeSyntax p &&
-                p.Keyword.Kind() == SyntaxKind.VoidKeyword;
+            return true;
```

Comment:
none of hte above logic was relevant.  we already ensured that all return-statements were not returning an expression (meaning they're all of the form `return;`).  In which case one of two things are true:

1. we're in a non-expr-returning function-like construct, which is fine.
2. we're in a expr-returning function, and hte code is in error, which is also fine, we can proceed just fine, since we go into 'best effort mode' anyways and have already warned in that case.

In other words, we want to warn when the user's code was correct, *but* we don't know how to extract properly.